lib/runner.coffee

```diff
@@ -50,11 +50,17 @@ class Runner
     cwd = @scriptOptions.workingDirectory
 
     workingDirectoryProvided = cwd? and cwd isnt ''
-    paths = atom.project.getPaths()
-    if not workingDirectoryProvided and paths?.length > 0
+    if not workingDirectoryProvided
       try
-        cwd = if fs.statSync(paths[0]).isDirectory() then paths[0] else path.join(paths[0], '..')
-    
+        switch atom.config.get('script.cwdBehavior')
+          when 'First project directory'
+            paths = atom.project.getPaths()
+            if paths?.length > 0
+              cwd = if fs.statSync(paths[0]).isDirectory() then paths[0] else path.join(paths[0], '..')
+          when 'Project directory of the script'
+            cwd = @getProjectPath()
+          when 'Directory of the script'
+            cwd = atom.workspace.getActivePaneItem().buffer.file.getParent().path
```

Comment:
i think, you should add some `?` (maybe with `or ''`) like [here](https://github.com/rgbkrk/atom-script/blob/master/lib/grammars.coffee#L135).